src/lib.rs

```diff
@@ -285,11 +285,23 @@ impl<'a> CSSInliner<'a> {
             }
         }
         if self.options.load_remote_stylesheets {
-            for link_tag in document
+            let mut links = document
                 .select("link[rel~=stylesheet]")
                 .map_err(|_| error::InlineError::ParseError(Cow::from("Unknown error")))?
-            {
-                if let Some(href) = link_tag.attributes.borrow().get("href") {
+                .map(|link_tag| {
+                    link_tag
+                        .attributes
+                        .borrow()
+                        .get("href")
+                        .as_ref()
+                        .unwrap_or_else(|| &"")
+                        .to_string()
+                })
```

Comment:
I think we can avoid some allocations & `is_empty` check with `filter_map`:
```suggestion
                .filter_map(|link_tag| link_tag.attributes.borrow().get("href").map(str::to_string))
```